-add fluent validation when adding employee and updating employee

AndreeaMonica opened the pull request:
-get all employees from database

AndreeaMonica committed:
-added fluent validation when adding employee and updating employee
-get all employees from database

moisoiu commented on HumanResources/Repositories/EmployeeRepository.cs:
```diff
@@ -51,11 +56,26 @@ public bool DeleteEmployee(long id)
+            if (!employees.Any())
+            {
+                throw new Exception("No Employees found");
+            }
```
When you get employees and you don't find any try to return an empty list or just do a **return employees** and treat the error in controller like 
If(!employees.Any())
{
return NoContent();
}
Return Ok(employees);

In this way you can treat it as a Server Response that there didn't happen something bad

moisoiu commented on HumanResources/Repositories/EmployeeRepository.cs:
```diff
@@ -51,11 +56,26 @@ public bool DeleteEmployee(long id)
         public long UpdateEmployee(long id, UpdateEmployeeRequest updateEmployee)
         {
             var employee = context.Employees.FirstOrDefault(x => x.Id == id);
+            if(employee == null)
```
Here is OK because you're doing an update to an entity and if you don't find it you can continue further, and even here you should change the Exception to something different, because Exception its too Generic. Try maybe ValidationException from FluentValidation or even better, because the new code say that throwing an Exception it's very resource consuming try something like bellow.

If(employee == null)
{
return -1 ;
}

And In controller you can do something
If (employeeId == -1) 
{
Return BadRequest("No Employee was updated. Employee not found");
}

moisoiu commented on HumanResources/Repositories/EmployeeRepository.cs:
```diff
@@ -41,7 +46,7 @@ public bool DeleteEmployee(long id)
             var employee = context.Employees.FirstOrDefault(x => x.Id == id);
             if(employee == null)
```
Please use the same approach as below comments

moisoiu commented on HumanResources/Repositories/EmployeeRepository.cs:
```diff
@@ -32,6 +33,10 @@ public long AddEmployee(AddEmployeeRequest employee)
         public GetEmployeeResponse GetEmployee(long id)
         {
             var employee = context.Employees.FirstOrDefault(x => x.Id == id);
+            if (employee == null)
```
Please use the same approach as below comments

AndreeaMonica committed:
Changes applied:
-errors are treated in controller as a Server Response

AndreeaMonica committed:
Applied changes:
-errors are treated in controller as a Server Response
-renamed variables in Controller
